packages/core/src/tracing/idleSpan.ts

```diff
@@ -209,9 +224,25 @@ export function startIdleSpan(startSpanOptions: StartSpanOptions, options: Parti
       return;
     }
 
-    const attributes = spanJSON.data || {};
-    if (spanJSON.op === 'ui.action.click' && !attributes[FINISH_REASON_TAG]) {
-      span.setAttribute(FINISH_REASON_TAG, _finishReason);
+    const attributes: SpanAttributes = spanJSON.data || {};
+    if (spanJSON.op === 'ui.action.click' && !attributes[SEMANTIC_ATTRIBUTE_SENTRY_IDLE_SPAN_FINISH_REASON]) {
+      span.setAttribute(SEMANTIC_ATTRIBUTE_SENTRY_IDLE_SPAN_FINISH_REASON, _finishReason);
+    }
+
+    // Save finish reason as tag, in addition to attribute, to maintain backwards compatibility
+    const scopes = getCapturedScopesOnSpan(span);
+
+    // Make sure to fetch up-to-date data here, as it may have been updated above
+    const finalAttributes: SpanAttributes = spanToJSON(span).data || {};
+    const finalFinishReason = finalAttributes[SEMANTIC_ATTRIBUTE_SENTRY_IDLE_SPAN_FINISH_REASON];
+    if (scopes.scope && typeof finalFinishReason === 'string') {
+      // We only want to add the tag to the transaction event itself
+      scopes.scope.addEventProcessor(event => {
+        if (event.type === 'transaction') {
+          event.tags = { ...event.tags, [FINISH_REASON_TAG]: finalFinishReason };
```

Comment:
Why exactly are we adding this back. Are we assuming people depend on this tag?

Reply:
I would also like to get rid of this, but in the spirit of not blocking this PR on aligning with performance folks about this, I'd rather add it for now and then we remove it when we know we can!

I guess the main impact would be it is not searchable anymore, which _may_ be relevant I think 🤔

Reply:
Right. Sounds good then!

Reply:
IMO we can remove this tag and make it span data, no reason for this to be indexed.

Reply:
ok, then I'll remove this! 👍